src/containers/FrameSelection/FrameSearchContainer.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { Dispatch } from 'redux';
+import { connect } from 'react-redux';
+import { ApplicationState } from '../../store';
+import { Boss } from '../../store/order/types';
+import { Frame } from '../../store/frames/types';
+import * as searchActions from '../../store/search/actions';
+import { frameSearchConfig } from '../../constants/frameSearch';
+import FrameSearch from '../../components/FrameSearch';
+
+type PropsFromState = {
+  isFetching: boolean;
+  errors: string;
+  boss: Boss;
+  frames: Frame[];
+  selectedFrames: Frame[];
+};
+
+type PropsFromDispatch = {
+  searchFrames: typeof searchActions.searchStart,
+  setSelectedFrames: typeof searchActions.setSelectedFrames,
+};
+
+type ComponentProps = PropsFromState & PropsFromDispatch;
+
+type StateProps = {
+  open: boolean;
+  query: string;
+};
+
+const mapStateToProps = (state: ApplicationState) => ({
+  isFetching: state.search.isFetching,
+  errors: state.search.errors,
+  boss: state.order.boss,
+  frames: state.search.frames,
+  selectedFrames: state.search.selectedFrames,
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  searchFrames: (boss: Boss, upc: string) => dispatch(searchActions.searchStart(boss, upc)),
+  setSelectedFrames: (frames: Frame[]) => dispatch(searchActions.setSelectedFrames(frames)),
+});
+
+class FrameSearchContainer extends React.Component<ComponentProps> {
+  state: StateProps = { open: false, query: '' };
+
+  private onChange = (props: Frame[]) => {
+    const { setSelectedFrames } = this.props;
+
+    setSelectedFrames(props);
+
+    return props;
+  }
+
+  private onBlur = (props: React.FocusEvent<HTMLElement>) => {
+    const { open } = this.state;
+
+    if (open) {
+      this.setState({ open: false })
+    }
+  }
+
+  private onInputChange = (props: string) => {
+    const { boss, searchFrames } = this.props;
+    const { query } = this.state;
+
+    if (props.length === frameSearchConfig.queryLength) {
+      if (query.length === 0 || query.indexOf(props)) {
```

Comment:
nice to move this to named constant to make people easier understand what happening here